app/backend/src/core/controllers/appraiser/BlockAppraiser.test.ts

```diff
@@ -0,0 +1,189 @@
+import Logger from "@/tools/Logger";
+import { BlockRewardsHandler } from "@/core/controllers/appraiser/handlers/BlockRewardsHandler";
+import { PriceService } from "@/core/controllers/appraiser/services/PriceService";
+import { getConfig } from "@/config";
+import { expect, mockFn } from "earl";
+import {
+  Block,
+  IBlockchainClient,
+} from "@/core/clients/blockchain/IBlockchainClient";
+import { BlockAppraiser } from "@/core/controllers/appraiser/BlockAppraiser";
+import {
+  WhitelistedAsset,
+  whitelistedMap,
+} from "@/core/clients/coingecko/assets/types";
+import StarknetClient from "@/core/clients/blockchain/starknet/StarknetClient";
+import { TransferService } from "@/core/controllers/appraiser/services/TransferService";
+import { TokenTransferHandler } from "@/core/controllers/appraiser/handlers/TokenTransfers";
+import { NativeTransferHandler } from "@/core/controllers/appraiser/handlers/NativeTransfers";
+
+describe(`${BlockAppraiser.name} on Starknet`, () => {
```

Comment:
This one is not exactly a unit, but more like an integration test. The only thing it mocks is the pricing service, since we don't have price history for testing.

Idea is that one can select any transaction from the explorer, estimate easily estimate its "value" given what he sees and add an test specifically for this transaction.

Drawback is that test requires internet connection and is a bit slower than others. I first went with mocking `StarknetClient` underlying provider responses to avoid this, but this makes new cases harder to create, read and maintain.